### PR TITLE
feature/9304 Add sidekiq and sidekiq cron

### PR DESCRIPTION
**Pull request by matthiasfehr**

### CODE REVIEW

File: ruby_on_rails/sidekiq.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Sidekiq
+
+* Add the following gem `gem 'sidekiq'` and install it
```
Comment (coorasse) [inline ruby_on_rails/sidekiq.md]: I would explain quickly what is this section about (background jobs) and the fact that we use sidekiq instead of suckerpunch because suckerpunch causes memory issues on Heroku and add a link to the github page of the gem

File: ruby_on_rails/sidekiq.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+config.active_job.queue_adapter = :inline
+```
+
+* configure the active job adapter in `config/environments/test.rb`
+
+```rb
+config.active_job.queue_adapter = :test
+```
+
+* create file `config/sidekiq.yml`
```
Comment (coorasse) [inline ruby_on_rails/sidekiq.md]: create a file and specify that this is an example but may change depending on specific configuration

File: ruby_on_rails/sidekiq.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+```yml
+:concurrency: 3
+:queues:
+ - [mailers, 1]
+ - [default, 1]
+```
+
+* on Heroku you need to:
+  * Add the Heroku [Redis addon](https://elements.heroku.com/addons/heroku-redis)
+  * Add env variable `REDIS_PROVIDER` = `REDIS_URL`
```
Comment (coorasse) [inline ruby_on_rails/sidekiq.md]: why is this needed?

File: ruby_on_rails/sidekiq.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  * And to start it turn on the worker
+
+* to run it locally you need to:
+  * Install Redis `brew install redis`
```
Comment (coorasse) [inline ruby_on_rails/sidekiq.md]: if not yet installed. can you please specify that? shall we have it installed in the laptop setup guide @schmijos ?

File: ruby_on_rails/sidekiq.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  * Start Redis `redis-server`
+  * Start Sidekiq `bundle exec sidekiq -C config/sidekiq.yml`
+  * Start your server
+
+## Sidekiq cron
+
+If the Heroku scheduler doesn't fit your needs, you can add extended capabilities
```
Comment (coorasse) [inline ruby_on_rails/sidekiq.md]: In which cases, for example, may not fit our needs?

File: ruby_on_rails/sidekiq.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+with the Sidekiq cron gem
```
Comment (coorasse) [inline ruby_on_rails/sidekiq.md]: link to the github page of the gem

File: ruby_on_rails/sidekiq.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+```rb
+schedule_file = 'config/schedule.yml'
+
+if File.exist?(schedule_file) && Sidekiq.server?
+  Sidekiq::Cron::Job.load_from_hash YAML.load_file(schedule_file)
+end
+```
+
+* create file `config/schedule.yml`
```
Comment (coorasse) [inline ruby_on_rails/sidekiq.md]: specify that this is an example and point to the gem README for complete instructions

**Comment by coorasse** [comment]
I actually wanted to reject it but I approved by mistake. Anyway, please take a look and assign it back to me afterwards :)

File: ruby_on_rails/sidekiq.md (hunk 3 above)
Comment (matthiasfehr) [reply ruby_on_rails/sidekiq.md]: read about it here, actually not sure if it really needs it
 https://github.com/mperham/sidekiq/wiki/Using-Redis#using-an-env-variable
